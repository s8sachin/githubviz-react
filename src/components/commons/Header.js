import React, { Component } from 'react';
import { Navbar, Nav, NavItem, MenuItem, NavDropdown, Button } from 'react-bootstrap';
import browserHistory from '../../history';

class Header extends Component {

  state = {
    userLoggedIn: false
  }
  
  componentWillMount () {
    this.setState({userLoggedIn: this.authenticate()})
  }

  authenticate () {
    return (localStorage.getItem('access_token') && localStorage.getItem('email')) ? true : false;
  }

  logout () {
    localStorage.removeItem('access_token');
    localStorage.removeItem('email');
    this.setState({userLoggedIn: false});
    browserHistory.push('/');
  }

  render () {
    return (
      <Navbar inverse collapseOnSelect>
        <Navbar.Header>
          <Navbar.Brand>
            <a href='#' id ="gitviz" onClick={() => browserHistory.push('/')}>Github Viz</a>
          </Navbar.Brand>
          <Navbar.Toggle />
        </Navbar.Header>
        <Navbar.Collapse>
          {this.state.userLoggedIn && 
            <div>
              <Nav>
                <NavItem eventKey={1} href="#">
                  <Button  bsSize='xsmall' bsStyle="primary" onClick={() => browserHistory.push('/allGraphs')}>All Graphs</Button>
                </NavItem>
              </Nav>
              <Nav>
                <NavItem eventKey={2} href="#">
                  <Button  bsSize='xsmall' bsStyle="primary" onClick={() => browserHistory.push('/googleMap')}>Google Map</Button>
                </NavItem>
                {/* <NavItem eventKey={2} href="#">
                  Link
                  </NavItem>
                  <NavDropdown eventKey={3} title="Dropdown" id="basic-nav-dropdown">
                  <MenuItem eventKey={3.1}></MenuItem>
                  <MenuItem eventKey={3.2}>Another action</MenuItem>
                  <MenuItem eventKey={3.3}>Something else here</MenuItem>
                  <MenuItem divider />
                  <MenuItem eventKey={3.3}>Separated link</MenuItem>
                </NavDropdown> */}
              </Nav>
            </div>
          }
          <Nav pullRight>
            <NavItem eventKey={1} id="piechart" onClick={() => browserHistory.push('/piecharts')}>
             Pie chart
            </NavItem>
            <NavItem  id="linegrph" eventKey={2} onClick={() => browserHistory.push('/lineGraph')}>
              Line Gragh
            </NavItem>
            {this.state.userLoggedIn && <NavItem eventKey={2}>
              <Button bsSize='xsmall' bsStyle="primary" onClick={() => this.logout()}>Logout</Button>
            </NavItem>}
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    )
  }
}

export default Header;