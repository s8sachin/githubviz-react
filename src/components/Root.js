import React, { Component } from 'react';
import { Grid, Jumbotron, Button } from 'react-bootstrap';
import Header from './commons/Header';

class Root extends Component {

  clickFunc() {
    console.log("hi")
  }
  render() {
    return (
      <div>
        <Header/>
        <Grid bsClass='container'>
          <Jumbotron>
            <h1>Hello, world!</h1>
            <p>
              This is a simple hero unit, a simple jumbotron-style component for calling
              extra attention to featured content or information.
            </p>
            <p>
              <Button bsStyle="primary" href='#'>Learn more</Button>
            </p>
          </Jumbotron>
        </Grid>
      </div>
    );
  }
}

export default Root;