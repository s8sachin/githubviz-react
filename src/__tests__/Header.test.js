import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Header from '../components/commons/Header';
import Root from '../components/Root';
import LocalStorageMock from '../LocalStorageMock';
import { Link } from 'react-router';
import PieCharts from '../components/piecharts';

global.localStorage = new LocalStorageMock;


configure({ adapter: new Adapter() });

describe('Header', () => {
it('should mount Header component', () => {
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })
   it("should click create github viz link", () => {
    const wrapper = mount(<Header />)
    // console.log(wrapper.instances());
    // expect(wrapper.instances().props.name).toBe('Header');
    // wrapper.find('abc').simulate('click');
    const tag = wrapper.find('#abc')
    // console.log(tag.simulate('click'))
    const output= tag.simulate('click')
    console.log(output)
    console.log(tag)

    // const component = renderer.create(tag.props().onClick)
    const tree = shallow(<Root />)
    expect(toJson(tree)).toMatchSnapshot();
  })

//   it("should click create piechart link", () => {
//     const wrapper = mount(<Header />)
//     const tag1 = wrapper.find('#my_id')
//     // tag1.simulate('click')
//     // const tree1 = shallow(<PieCharts />)
//     // expect(toJson(tree1)).toMatchSnapshot();
//   })





   it("should click create piechart link", () => {
    const wrapper = shallow(<Header />)
    const button = wrapper.find('Navbar').find('Navbar.Collapse').find('Nav').find('NavItem').first()
    // const component = renderer.create(button.props().onClick)
    // let tree = component.toJSON();
    // tree.props.onClick();
    // expect(tree).toMatchSnapshot();
  })



//   it("should click create a allgraph ", () => {
//     const wrapper = shallow(<Header />)
//    console.log(wrapper.instances());
//     // const Button = wrapper.find('Navbar').find('Nav').first().find('Button')
//     const Button = wrapper.find('my_id')
    
//     console.log(Button);
    // const component = renderer.create(card.props().link)
  //   let tree = component.toJSON();
  //   tree.props.onClick();
  //   expect(tree).toMatchSnapshot();
//   })

// it("should click create github viz link", () => {
//     const wrapper = render(<Header />)
//     // console.log(wrapper.instances());
//     // expect(wrapper.instances().props.name).toBe('Header');
//     // const tag = wrapper.find('Navbar').find('Navbar.Brand').first().find('a')
//     // const component = renderer.create(tag.props().href)
//     expect(wrapper.find(Link).props().to).toBe('/Header');
//     let tree = component.toJSON();
//     tree.props.onClick();
//     expect(tree).toMatchSnapshot();
//   })



// it("should click create github viz link", () => {
//     const wrapper = mount(<Header />)
//     // console.log(wrapper.instances());
//     // expect(wrapper.instances().props.name).toBe('Header');
//     // wrapper.find('abc').simulate('click');
//     const tag = wrapper.find('#abc')
//     tag.simulate('click')
//     // const component = renderer.create(tag.props().href)
//     // let tree = component.toJSON();
//     // tree.props.onClick();
//     expect(tag).toMatchSnapshot();
//   })
})