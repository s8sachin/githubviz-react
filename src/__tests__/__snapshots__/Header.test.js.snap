// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should click create github viz link 1`] = `
<div>
  <Header />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <Jumbotron
      bsClass="jumbotron"
      componentClass="div"
    >
      <h1>
        Hello, world!
      </h1>
      <p>
        This is a simple hero unit, a simple jumbotron-style component for calling extra attention to featured content or information.
      </p>
      <p>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          disabled={false}
          onClick={[Function]}
        >
          Login
        </Button>
      </p>
    </Jumbotron>
  </Grid>
</div>
`;

exports[`Header should mount Header component 1`] = `
<Uncontrolled(Navbar)
  bsStyle="default"
  collapseOnSelect={true}
  inverse={true}
>
  <NavbarHeader>
    <NavbarBrand>
      <a
        href="#"
        id="abc"
        onClick={[Function]}
      >
        Github Viz
      </a>
    </NavbarBrand>
    <NavbarToggle />
  </NavbarHeader>
  <NavbarCollapse>
    <Nav
      bsClass="nav"
      justified={false}
      pullLeft={false}
      pullRight={true}
      stacked={false}
    >
      <NavItem
        active={false}
        disabled={false}
        eventKey={1}
        id="my_id"
        onClick={[Function]}
      >
        Pie chart
      </NavItem>
      <NavItem
        active={false}
        disabled={false}
        eventKey={2}
        onClick={[Function]}
      >
        Line Gragh
      </NavItem>
    </Nav>
  </NavbarCollapse>
</Uncontrolled(Navbar)>
`;
